sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -602,15 +609,55 @@ private[lf] object SBuiltinFun {
       SText(cctp.MessageDigest.digest(Ref.HexString.assertFromString(getSText(args, 0))))
   }
 
-  final case object SBSECP256K1Bool extends SBuiltinPure(3) {
-    override private[speedy] def executePure(args: util.ArrayList[SValue]): SBool = {
+  final case object SBSECP256K1Bool extends SBuiltinFun(3) {
+    private[speedy] def execute[Q](
+        args: util.ArrayList[SValue],
+        machine: Machine[Q],
+    ): Control[Q] = {
       val signature = Ref.HexString.assertFromString(getSText(args, 0))
       val digest = Ref.HexString.assertFromString(getSText(args, 1))
-      val publicKey = extractPublicKey(Ref.HexString.assertFromString(getSText(args, 2)))
 
-      SBool(cctp.MessageSignature.verify(signature, digest, publicKey))
+      try {
+        val publicKey = extractPublicKey(Ref.HexString.assertFromString(getSText(args, 2)))
+
+        Control.Value(SBool(cctp.MessageSignature.verify(signature, digest, publicKey)))
+      } catch {
+        case _: NoSuchProviderException =>
+          crash("JCE Provider BouncyCastle not found")
+        case _: NoSuchAlgorithmException =>
+          crash("BouncyCastle provider fails to support SECP256K1")
+        case _: InvalidKeyException =>
+          // TODO: https://github.com/DACH-NY/canton-network-utilities/issues/2916: introduce structured CCTP error reporting
+          //  Control.Error(
+          //    IE.Dev(
+          //      NameOf.qualifiedNameOfCurrentFunc,
+          //      IE.Dev.CCTP(IE.Dev.CCTP.InvalidKeyError(exn.getMessage)),
+          //    )
+          //  )
+          Control.Value(SBool(false))
```

Comment:
For the moment, we simply return false rather than reporting a more structured error. This is done to avoid to reduce the changes blast radius.